MenuXDGCheck.py

```diff
@@ -25,7 +25,44 @@ def __init__(self):
         # $ echo $XDG_DATA_DIRS/applications
         # /var/lib/menu-xdg:/usr/share
         AbstractCheck.AbstractFilesCheck.__init__(
-            self, "MenuXDGCheck", r"/usr/share/applications/.*\.desktop$")
+            self, "MenuXDGCheck", r'(?:/usr|/etc/opt|/opt/.*)/share/applications/.*\.desktop$')
+
+    def parse_desktop_file(self, pkg, root, f, filename):
+        cfp = cfgparser.RawConfigParser()
+        try:
+            cfp.read(f)
+        except cfgparser.DuplicateSectionError as e:
+            printError(
+                pkg, 'desktopfile-duplicate-section', filename,
+                '[%s]' % e.section)
+        except cfgparser.DuplicateOptionError as e:
+            printError(
+                pkg, 'desktopfile-duplicate-option', filename,
+                '[%s]/%s' % (e.section, e.option))
+        except cfgparser.MissingSectionHeaderError:
+            printError(
+                pkg, 'desktopfile-missing-header', filename)
+        except (cfgparser.ParsingError, UnicodeDecodeError) as e:
+            printWarning(
+                pkg, 'invalid-desktopfile', filename)
```

Comment:
I think this deserves some information what made the desktopfile invalid, i.e. something from `e` in the message.

Reply:
the error isn't particularly helpful, it always says something like "Parse error while handling file: XYZ". I've included it.